scalyr_agent/sharded_copying_manager/copying_manager.py

```diff
@@ -1137,20 +1192,66 @@ def _scan_for_new_logs_if_necessary(
             log_matchers, checkpoints=checkpoints, copy_at_index_zero=copy_at_index_zero
         )
 
+    def __prepare_checkpoints(self):
+        """
+        Combines all worker checkpoints that were left from the previous agent run.
+        It combines checkpoint files into one single file in order to
+        allow workers to rewrite the content of this files without losing any data.
+        """
+
+        checkpoints_dir_path = os.path.join(
+            self.__config.agent_data_path, "checkpoints"
+        )
+
+        # if there is a file called as the checkpoints folder, remove it
+        if os.path.isfile(checkpoints_dir_path):
+            os.unlink(checkpoints_dir_path)
+
+        # create new checkpoint folder if does not exist.
+        if not os.path.exists(checkpoints_dir_path):
+            os.mkdir(checkpoints_dir_path)
+
+        # read all checkpoints from the manager's previous run and combine them into one mater file.
+        checkpoints = self.__find_and_read_checkpoints()
+
+        master_checkpoints_path = os.path.join(
+            checkpoints_dir_path, CHECKPOINTS_MASTER_FILE_NAME,
+        )
+        write_checkpoint_state_to_file(
+            checkpoints, master_checkpoints_path, time.time()
+        )
+
+        # clear the checkpoints folder by removing all worker checkpoint files.
+        checkpoints_glob = os.path.join(checkpoints_dir_path, "*checkpoints*.json")
+
+        # also remove old version checkpoints
+        for path in glob.glob(
+            os.path.join(self.__config.agent_data_path, "*checkpoints.json")
```

Comment:
Do you need this separate glob check?  I believe `*checkpoints*.json` will also match `*checkpoints.json`.  I think we could simplify the code and just do the glob for `*checkpoints*.json`.

Reply:
Yes, that makes things easier, thanks.